pkg/debugapi/wallet.go

```diff
@@ -14,10 +14,10 @@ import (
 )
 
 type walletResponse struct {
-	BZZ             float64        `json:"bzz"`
-	XDai            float64        `json:"xDai"`
-	ChainID         int64          `json:"chainID"`
-	ContractAddress common.Address `json:"contractAddress"`
+	BZZ             float64        `json:"bzz"`             // the BZZ balance of the wallet associated with the eth address of the node
```

Comment:
In general it's not a good idea to use `float` for representing monetary values, because at small amounts it starts to lose precision. There are some languages which has `bigint` or `decimal` types for storing that, unfortunately in JSON there is no such types.

Because of this previously with the `/balance` endpoint we used `string` type to be able encode the value of the internal `bigint` type for expressing that in a lossless way: https://docs.ethswarm.org/debug-api/#tag/Balance/paths/~1balances/get

Reply:
This raises another question: we can return the balance with either big or small units, meaning: if the balance is 17 BZZ, we can return 17 or 170000000000000000 (which is what the erc20 service returns). I've already wrote the logic for converting the small units balance for better UX.

Reply:
I would put my vote in for consistency and always return amounts as integers (or strings) relative to the smallest possible unit of measurement.  In other words, mirror what the erc20 and other blockchain APIs do.

Reply:
Agree with @ldeffenb here.